### ISSUE TITLE
EAS development build failing for Android (expo sdk 45)

### ISSUE THREAD

**Issue by antoinebrtd** [expo/eas-cli#1238]
### Build/Submit details page URL

https://expo.dev/accounts/antoine92/projects/ifc-app/builds/45c58c53-ba84-4366-9402-8d024e1c6f0e

### Summary

I am trying to run `eas build -p android --profile development` which is failing, and I am able to reproduce what I see on the build page by running locally `expo run:android --variant release` using following java version:
```
openjdk 11.0.15 2022-04-19
OpenJDK Runtime Environment (build 11.0.15+10-Ubuntu-0ubuntu0.20.04.1)
OpenJDK 64-Bit Server VM (build 11.0.15+10-Ubuntu-0ubuntu0.20.04.1, mixed mode, sharing)
```

### Managed or bare?

Managed worflow

### Environment

```
$ npx expo-env-info

    expo-env-info 1.0.5 environment info:
        System:
          OS: Linux 5.4 Ubuntu 20.04.4 LTS (Focal Fossa)
          Shell: 5.8 - /usr/bin/zsh
        Binaries:
          Node: 16.9.1 - ~/.nvm/versions/node/v16.9.1/bin/node
          npm: 7.21.1 - ~/.nvm/versions/node/v16.9.1/bin/npm
        npmPackages:
          expo: ^45.0.0 => 45.0.6 
          react: 17.0.2 => 17.0.2 
          react-dom: 17.0.2 => 17.0.2 
          react-native: 0.68.2 => 0.68.2 
          react-native-web: 0.17.7 => 0.17.7 
        npmGlobalPackages:
          eas-cli: 0.55.1
          expo-cli: 6.0.0
        Expo Workflow: managed
``` 

```
$ expo doctor

    Expected package @expo/config-plugins@^4.1.0
    Found invalid:
      @expo/config-plugins@5.0.0
      (for more info, run: npm why @expo/config-plugins)
```

### Error output

```
$ expo run:android --variant release

  › Building app...
  Configuration on demand is an incubating feature.
  
  > Configure project :app
  Unable to detect AGP versions for included builds. All projects in the build should use the same AGP version. Class name for the included build object: org.gradle.composite.internal.DefaultIncludedBuild$IncludedBuildImpl_Decorated.
  
  FAILURE: Build completed with 2 failures.
  
  1: Task failed with an exception.
  -----------
  * Where:
  Build file '/home/antoine/Projects/IFC-App/android/app/build.gradle' line: 92
  
  * What went wrong:
  A problem occurred evaluating project ':app'.
  > Could not read script '/' as it is a directory.
  
  * Try:
  > Run with --stacktrace option to get the stack trace.
  > Run with --info or --debug option to get more log output.
  > Run with --scan to get full insights.
  ==============================================================================
  
  2: Task failed with an exception.
  -----------
  * What went wrong:
  A problem occurred configuring project ':app'.
  > compileSdkVersion is not specified. Please add it to build.gradle
  
  * Try:
  > Run with --stacktrace option to get the stack trace.
  > Run with --info or --debug option to get more log output.
  > Run with --scan to get full insights.
  ==============================================================================
  
  * Get more help at https://help.gradle.org
  
  BUILD FAILED in 9s
  6 actionable tasks: 6 up-to-date
  /home/antoine/Projects/IFC-App/android/gradlew exited with non-zero code: 1
```

### Reproducible demo or steps to reproduce from a blank project

Not sure how to reproduce. This is an entire expo project which is several years old. Did not have any issues upgrading from sdk 44 to 45 recently, and tried eas build on expo 44 and it is failing as well. I provided a simple snack, containing my root files, including my package-json and my eas configuration.
https://snack.expo.dev/@antoine92/34a53b

**Comment by dsokal** [comment]
It seems this is not a problem with EAS Build but rather with your project's configuration. 

Most likely the problem is in `build.gradle`:
```
  * Where:
  Build file '/home/antoine/Projects/IFC-App/android/app/build.gradle' line: 92
  
  * What went wrong:
  A problem occurred evaluating project ':app'.
  > Could not read script '/' as it is a directory.
```

**Comment by antoinebrtd** [comment]
@dsokal thank you for your comment. I am just following the eas expo guide to [create my first build](https://docs.expo.dev/build/setup/). Section 4 is not working, [eas build -p android --profile development](https://docs.expo.dev/build-reference/apk/) is failing and to trouble shoot it I built my app locally using `expo run:android --variant release`. Expo is generating the build.gradle file locally, I did not manually generate any native code, since my app is under managed workflow
